### ISSUE TITLE
No public way to ask whether a tensor is on the autograd tape, which makes no_grad untestable downstream

### ISSUE THREAD

**Issue by vyncint** [vyncint/oxmera#20]
## Summary

`Tensor::requires_grad()` answers "will gradients accumulate **on this tensor**" — leaf-ness — not "is this tensor part of the graph". That is a deliberate choice and the doc comment says so. The gap is that the *other* predicate, `is_tracked()`, is `pub(crate)`, so downstream code has no way to ask the second question at all.

## The trap

The PyTorch-shaped assertion a user will reach for is silently vacuous:

```rust
let a = Tensor::from_slice(&[1.0, 2.0], [1, 2]).unwrap().requires_grad_(true);

let inside  = oxmera::no_grad(|| a.mul_scalar(3.0).unwrap()).requires_grad(); // false
let outside = a.mul_scalar(3.0).unwrap().requires_grad();                     // ALSO false
```

Both are `false`, so `assert!(!y.requires_grad())` passes whether or not `no_grad` did anything. I wrote exactly that test while exercising the crate and it passed for the wrong reason.

To be clear about what is and is not broken: **`no_grad` itself works correctly**, and it is properly thread-local. Verified indirectly, which is currently the only way:

```rust
oxmera::no_grad(|| a.mul_scalar(3.0).unwrap().sum(&[0,1]).unwrap().backward().unwrap());
assert!(a.grad().is_none());   // passes — the tape really was suppressed
```

That workaround requires running a whole backward pass and inspecting a *different* tensor, which is not a reasonable way to write an assertion.

## Proposal

Make the predicate public — either by exposing the existing method or under a name that cannot be confused with `requires_grad`:

```rust
/// Whether this tensor participates in the autograd tape — true for
/// leaves that require grad and for anything computed from one while
/// recording was enabled.
pub fn is_tracked(&self) -> bool
```

and add a sentence to the `requires_grad` docs contrasting the two, since the difference from PyTorch (`y = x * 2; y.requires_grad == True`) will surprise most people arriving from there.

## Why it matters beyond tidiness

- Downstream test suites cannot assert that an inference path is allocation-free of tape nodes.
- Library authors wrapping oxmera cannot branch on trackedness to skip recording work.
- The natural test of the feature is vacuous, so a regression in `no_grad` would not be caught by user code that looks like it is testing it.
